Robot2019/src/main/java/frc/robot/lib/logging/LogFiles.java

```diff
@@ -0,0 +1,186 @@
+        }
+    }
+
+    private static void findExistingLogIds() {
+        existingLogIds.clear();
+        File[] files = dirFile.listFiles();
+        for(File file: files) {
+            String name = file.getName();
+            if(name.indexOf(".") == -1) {
+                continue;
+            }
+            String idString = file.getName().substring(0, name.indexOf("."));
+            try {
+                int id = Integer.parseInt(idString);
+                if(!existingLogIds.contains(id)) {
+                    existingLogIds.add(id);
+                }
+            } catch(NumberFormatException e) {
+                continue;
+            }
+        }
+    }
+
+    private static void clearOldLogs() throws AbortException {
+        int lines = 0;
+        ArrayList<String> linesD = new ArrayList<>();
+        try(BufferedReader br = new BufferedReader(new FileReader(infoFile))) {
+            String line;
+            while((line = br.readLine()) != null) {
+                linesD.add(line);
+                lines++;
+            }
+        } catch(IOException e) {
+            throw new AbortException("reading info file", e);
+        }
+        int endIdx = lines - 100;
+        if(lines > 100) {
+            try(BufferedWriter bw = new BufferedWriter(new FileWriter(infoFile))) {
+                for(int i = endIdx+1; i < lines; i++) {
+                    bw.append(linesD.get(i));
+                    bw.newLine();
+                }
+            } catch(IOException e) {
+                throw new AbortException("writing info file", e);
+            }
+            for(int i = 0; i < endIdx+1; i++) {
+                try {
+                    int id = Integer.parseInt(linesD.get(i).substring(0, linesD.get(i).indexOf(" ")));
+                    new File(dirString + "/" + id + ".txt").delete();
+                    new File(dirString + "/" + id + ".csv").delete();
+                    existingLogIds.remove(id);
+                } catch(Exception e) {}
+            }
+        }
+        linesD.clear();
+        try(BufferedReader br = new BufferedReader(new FileReader(infoFile))) {
+            String line;
+            while((line = br.readLine()) != null) {
+                linesD.add(line);
+            }
+        } catch(IOException e) {
+            throw new AbortException("reading info file", e);
+        }
+        try(BufferedWriter bw = new BufferedWriter(new FileWriter(infoFile))) {
+            for(int i = 0; i < lines; i++) {
+                String line = linesD.get(i);
+                try {
+                    if(!existingLogIds.contains(Integer.parseInt(line.substring(0, line.indexOf(" "))))) {
+                        continue;
+                    }
+                } catch(Exception e) {e.printStackTrace();}
+                bw.append(linesD.get(i));
+                bw.newLine();
+            }
+        } catch(IOException e) {
+            throw new AbortException("writing info file", e);
+        }
```

Comment:
Seems like you should be able to combine this code with the earlier code in this method to improve readability. It's counterintuitive to me to rewrite the info file twice.